File: scripts/inspect_excel.py
import pandas as pd
import os

file_path = r"C:\Users\kgene\Desktop\subtracker-V3-main\transactions.xlsx"
print(f"Reading file: {file_path}")

try:
    # Try reading with openpyxl engine explicit, usually required for xlsx
    df = pd.read_excel(file_path, engine='openpyxl')
    print("Headers:", list(df.columns))
    if not df.empty:
        print("First row:", df.iloc[0].to_dict())
    print("Total Rows:", len(df))
except ImportError:
    print("Error: openpyxl not installed? installing...")
    # This might fail if pip is not in path or permission denied, but worth a try or just fail and ask user
    print("Please ensure openpyxl is installed: pip install openpyxl")
except Exception as e:
    print(f"Error reading excel: {e}")

File: scripts/inspect_excel.js
import { createRequire } from 'module';
import path from 'path';
import { fileURLToPath } from 'url';

const require = createRequire(import.meta.url);
const XLSX = require('xlsx');

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const filePath = path.join(__dirname, '..', 'transactions.xlsx');

console.log(`Reading file: ${filePath}`);

try {
    const workbook = XLSX.readFile(filePath);
    const sheetName = workbook.SheetNames[0];
    console.log(`Sheet Name: ${sheetName}`);
    const worksheet = workbook.Sheets[sheetName];
    const data = XLSX.utils.sheet_to_json(worksheet);

    if (data.length > 0) {
        console.log('Headers:', Object.keys(data[0]));
        console.log('First row:', data[0]);
        console.log('Total rows:', data.length);
    } else {
        console.log('Sheet is empty');
    }
} catch (error) {
    console.error('Error reading file:', error.message);
}
